include/taskolib/GitObjectWrapper.h

```diff
@@ -0,0 +1,180 @@
+#include <git2.h>
+#include <utility>
+
+/**
+ * A wrapper to handle the raw C pointer of git_repository
+**/
+class GitRepositoryPtr
+
+{
+public:
+    // empty
+    //GitRepositoryPtr(){}
+
+    // normal
+    GitRepositoryPtr(git_repository* repo): repo_{repo} {};
+
+    // copy constructor
+    GitRepositoryPtr(const GitRepositoryPtr& gl): GitRepositoryPtr{gl.repo_} {};
+
+    // move constructor
+    GitRepositoryPtr(GitRepositoryPtr&& gl): repo_{std::exchange(gl.repo_, nullptr)} {};
+
+    // destructor
+    ~GitRepositoryPtr(){ git_repository_free(repo_); };
+
+    // copy assignment
+    GitRepositoryPtr& operator=(const GitRepositoryPtr& gl) {return *this = GitRepositoryPtr{gl};};
+
+    // move assignment
+    GitRepositoryPtr& operator=(GitRepositoryPtr&& gl) noexcept
+    {
+        std::swap(repo_, gl.repo_);
+        return *this;
+    }
+
+    git_repository** getptr() {return &repo_;};
+
+    void reset()
+    {
+        git_repository_free(repo_);
+        repo_ = nullptr;
+    }
+
+
+    //getter
+    git_repository* get() {return repo_;} ;
+    const git_repository* get() const {return repo_;} ;
+
+
+private:
+    git_repository* repo_;    
+};
+
+
+class GitSignaturePtr
+
+{
+public:
+
+    // empty
+    //GitSignaturePtr(){}
+
+    // normal
+    GitSignaturePtr(git_signature* signature): signature_{signature} {};
+
+    // copy constructor
+    GitSignaturePtr(const GitSignaturePtr& gl): GitSignaturePtr{gl.signature_} {};
+
+    // move constructor
+    GitSignaturePtr(GitSignaturePtr&& gl): signature_{std::exchange(gl.signature_, nullptr)} {};
+
+    // destructor
+    ~GitSignaturePtr(){ git_signature_free(signature_); };
+
+    // copy assignment
+    GitSignaturePtr& operator=(const GitSignaturePtr& gl) {return *this = GitSignaturePtr{gl};};
+
+    // move assignment
+    GitSignaturePtr& operator=(GitSignaturePtr&& gl) noexcept
+    {
+        std::swap(signature_, gl.signature_);
+        return *this;
+    }
+
+    git_signature** getptr() {return &signature_;};
+
+
+    //getter
+    git_signature* get() {return signature_;} ;
```

Comment:
... the same goes for function comments. We use one of two styles for our Doxygen comments:
```cpp
/**
 * For multi-line
 * comments.
 */

/// For single-line comments.
```

Documentation for functions should be written in present tense active, in this case:
```
/// Return a raw pointer to the git_signature pointer managed by the object.
```